web/book/src/internals/modules.md

```diff
@@ -0,0 +1,95 @@
+# Modules
+
+PRQL needs modules that would:
+
+- allow importing declarations from other files,
+- have namespaces for things like `std`,
+- have hierarchical structure so we can represent files in directories,
+- have unambiguous module structure within a project,
+- be able to compile individual files that are a part of a project.
+
+## Basic definition
+
+Module is a namespace that contains declarations.
+
+Modules can be defined with mod keyword and a code block encased in curly
+braces:
+
+```
+mod my_module {
+    let bangers = (from tracks | take 10)
+}
+```
+
+## File loading
+
+If a module definition of `my_module` does not have a code block, it's contents
+are loaded from `./my_module.prql`.
+
+If the file does not exist, contents are loaded from file
```

Comment:
I dislike this this, I think it make the module system more complicated, more to learn, more ways to do one thing, it increases the needs for best practices and to reach consensus.
I think it would better to have just one way, that it should always load `./my_module.prql` without any fallback.

I think having several files with the same filename (`mod.prql`) is a nightmare, it makes it difficult to search, to find files, to determine which file something is in.